tests/integration/test_install.py

```diff
@@ -0,0 +1,140 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test cylc install."""
+
+import pytest
+from pathlib import Path
+
+from .test_scan import init_flows
+
+from cylc.flow.async_util import pipe
+from cylc.flow.scripts import scan
+from cylc.flow.workflow_files import WorkflowFiles
+from cylc.flow.scripts.install import (
+    InstallOptions,
+    install_cli
+)
+
+
+SRV_DIR = Path(WorkflowFiles.Service.DIRNAME)
+CONTACT = Path(WorkflowFiles.Service.CONTACT)
+RUN_N = Path(WorkflowFiles.RUN_N)
+INSTALL = Path(WorkflowFiles.Install.DIRNAME)
+
+INSTALLED_MSG = "INSTALLED {wfrun} from"
+WF_ACTIVE_MSG = '1 run of "{wf}" is already active:'
+BAD_CONTACT_MSG = "Bad contact file:"
+
+
+class graphql_query_mock:
+    """Context manager to mock network.scan.graphql_query().
+
+    (In the scripts.scan namepace, where it is used below).
+
+    """
+    def __enter__(self):
+        """Replace the original function."""
+        @pipe
+        async def mock_graphql_query(flow, fields, filters=None):
+            """The fake function."""
+            flow.update({"status": "running"})
+            return flow
+        self.orig_func = scan.graphql_query
+        scan.graphql_query = mock_graphql_query
+
+    def __exit__(self, *args, **kwargs):
+        """Restore the original function."""
+        scan.graphql_query = self.orig_func
+
+
+@pytest.fixture()
+def src_run_dirs(mock_glbl_cfg, monkeypatch, tmp_path: Path):
+    """Create some workflow source and run dirs for testing.
+
+    Source dirs:
+      <tmp-src>/w1
+      <tmp-src>/w2
+
+    Run dir:
+      <tmp-run>/w1/run1
+
+    """
+    tmp_src_path = tmp_path / 'cylc-src'
+    tmp_run_path = tmp_path / 'cylc-run'
+    tmp_src_path.mkdir()
+    tmp_run_path.mkdir()
+
+    init_flows(
+        tmp_run_path=tmp_run_path,
+        running=('w1/run1',),
+        tmp_src_path=tmp_src_path,
+        src=('w1', 'w2')
+    )
+    mock_glbl_cfg(
+        'cylc.flow.workflow_files.glbl_cfg',
+        f'''
+            [install]
+                source dirs = {tmp_src_path}
+        '''
+    )
+    monkeypatch.setattr('cylc.flow.pathutil._CYLC_RUN_DIR', tmp_run_path)
+
+    return tmp_src_path, tmp_run_path
+
+
+def test_install_scan_no_ping(src_run_dirs, capsys, caplog):
+    """At install, running intances should be reported.
+
+    Ping = False case: don't query schedulers.
+    """
+
+    opts = InstallOptions()
+    opts.no_ping = True
+
+    install_cli(opts, reg='w1')
+    out = capsys.readouterr().out
+    assert INSTALLED_MSG.format(wfrun='w1/run2') in out
+    assert WF_ACTIVE_MSG.format(wf='w1') in out
+    assert f"{BAD_CONTACT_MSG} w1/run1" in caplog.text
```

Comment:
Why is this bad contact file message expected?

Reply:
Because the contact file has no content (it's faked with `touch`). I'll add a comment.

Reply:
Done.